### PR TITLE
Android keyboard input type fix

### PR DESCRIPTION
**Pull request by flogy**
On some devices (as my HTC One X plus running Android 4.2.2), the default keyboard that is used in libGDX always uses the input type that was last used. This means when you run some application that uses a digit-only keyboard, then exit that application and start the libGDX application directly, a digit-only keyboard will show up when trying to write into an InputField.
This fix uses an old way to handle keyboard input that was already built into libGDX, but never used. It just uses a hidden TextView element with the input type TEXT. The key inputs are then routed to the AndroidInput as usual.

Second commit is about password mode of TextFields. If the font of a text field does not contain the defined (or default) password character, it will currently just write plain text. There is no error or log message and could therefore be confusing, as it was for me. I suggest throwing an exception in that case.

### CODE REVIEW

**Comment by flogy** [comment]
The exception thingy is not important (was not supposed to get in this pull request). But what do you guys think about the other part? I mean, the current version of LibGDX is still not working properly on all devices because of this problem!

File: gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java [hunk 1]
```diff
@@ -348,7 +348,10 @@ void updateDisplayText () {
 		}
 		String newDisplayText = buffer.toString();
 
-		if (passwordMode && font.containsCharacter(passwordCharacter)) {
+		if (passwordMode) {
```
Comment (siondream) [inline gdx/src/com/badlogic/gdx/scenes/scene2d/ui/TextField.java]: Except for this, I'm okay with the PR.

**Comment by siondream** [comment]
Could you please remove the exception in the `TextField` and pull from upstream so we can merge this? Thanks!

**Comment by MobiDevelop** [comment]
Let's make sure that we test this (on a variety of devices with a variety of keyboard types) before we merge, please.

**Comment by badlogic** [comment]
@siondream @MobiDevelop can you sign this one off if it's OK? We can eliminate the exception ourselves i guess.

**Comment by Tom-Ski** [comment]
Aside from it not being able to automatically merge, this did not work on my device.  So just a heads-up in case some feels merge happy.

**Comment by MobiDevelop** [comment]
Since this can't be merged, doesn't work on Tomski's device, and needs changes, I'm going to close this. 

On a related note, we should address how to handle text editing without relying on key events since not all IMEs will send key events.

**Pull request closed without merging**